src/elm/interval.coffee

```diff
@@ -171,78 +260,67 @@ module.exports.Expand = class Expand extends Expression
       if !interval.low? or !interval.high?
         return null
 
-      # if we mix date and datetime, convert to datetime
-      if ((interval.low.isDate or interval.high.isDate) and 
-      (interval.low.isDateTime or interval.high.isDateTime))
-        interval = interval.copy()
+      if type is 'datetime' #support for implicitly converting dates to datetime
         interval.low = interval.low.getDateTime()
         interval.high = interval.high.getDateTime()
-      
-      # Support for Date, Datetime, Time
-      if ((interval.low.isDate and interval.high.isDate) or
-      (interval.low.isDateTime and interval.high.isDateTime) or
-      (interval.low.isTime?() and interval.high.isTime?()))
-        func = @expandDTishInterval
-
-      # Support for Quantity
-      else if (interval.low.isQuantity and interval.high.isQuantity)
-        func = @expandQuantityInterval
-
-      # Support for Decimal and Integer (note Decimals with .0 (eg 1.0) are considered Integers)
-      else if (typeof interval.low is 'number' and typeof interval.high is 'number')
-        func = @expandNumericInterval
-
-      if func?
-        items = func.call(@,interval,per)
-        return null if items == null
-        pushOntoArray(results,items)
-      # If we are here, Interval type is unsupported or mismatched
-      else
-        return null
+
+      per = per ? defaultPer(interval)
```

Comment:
Yes, I agree, and I think this new code will work that way, since per is declared outside the loop and should only be set the first time this line gets hit. I could have put the 'per' setting logic outside of the main loop, but it seemed like then it might be a bit funny with the open ended checks and the datetime implicit conversion.